src/telliot_feeds/sources/mimicry/mimicry_mashup_source.py

```diff
@@ -0,0 +1,207 @@
+
+from aiohttp import ClientConnectionError
+from aiohttp import ClientError
+from aiohttp import ClientResponseError
+from aiohttp import ClientSession
+from aiohttp import ClientTimeout
+from telliot_core.apps.telliot_config import TelliotConfig
+
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+
+# for coingecko api chain ids
+block_chains = {"ethereum-mainnet": "ethereum", "solana-mainnet": "solana"}
+
+
+@dataclass
+class NFTMashupSource(DataSource[str]):
+    """
+    DataSource for NFT market-cap mash up expected response data.
+    """
+
+    api_key = TelliotConfig().api_keys.find(name="nftgo")[0].key
+    metric: Optional[str] = None  # ("market_cap")
+    currency: Optional[str] = None  # ("usd")
+    collections: Optional[Any] = None  # ("chain-name", "contract-address")
+    tokens: Optional[Any] = None  # ("chain-name", "token-symbol", "contract-address")
+
+    async def fetch(self, url: str, headers: Optional[dict[str, str]]) -> Optional[Any]:
+        """
+        Fetches data from a given URL.
+
+        Args:
+            url (str): The URL to fetch data from.
+            headers (Optional[dict[str, str]]): Headers to send with the request.
+
+        Returns:
+            Response data from the API, or None if the request fails.
+
+        """
+        async with ClientSession(headers=headers) as session:
+            timeout = ClientTimeout(total=10)
+            async with session.get(url, timeout=timeout) as response:
+                return await response.json()
+
+    async def fetch_urls(self, urls: list[str], headers: Optional[dict[str, str]] = None) -> Any:
+        """
+        Fetches a list of URLs and returns a list of responses.
+
+        Args:
+            urls (list[str]): A list of URLs to fetch.
+            headers (Optional[dict[str, str]]): Headers to send with the request.
+
+        Returns:
+            Any: A list of responses.
+        """
+        tasks = [asyncio.create_task(self.fetch_url_with_retry(url=url, headers=headers)) for url in urls]
+        responses = await asyncio.gather(*tasks)
+        return responses
+
+    async def fetch_url_with_retry(
+        self, url: str, headers: Optional[dict[str, str]], retries: int = 3
+    ) -> Optional[Any]:
+        """
+        Fetches a response from an api with retry logic.
+
+        Args:
+            url (str): A single URL to fetch.
+            headers (Optional[dict[str, str]]): Headers to send with the request.
+
+        Returns:
+            Any: A list of responses.
+        """
+        for attempt in range(retries):
+            try:
+                response = await self.fetch(url=url, headers=headers)
+                return response
+            except (ClientError, ClientConnectionError, ClientResponseError) as e:
```

Comment:
Should add tests for these if expected. Also, what if it throws an error not included in this tuple?